### PR TITLE
Make concurrent stream writer enqueue methods return the size of events ingested

### PR DESCRIPTION
**Pull request by jguery**
This PR is part of the effort to reduce the bigger PR https://github.com/caskdata/cdap/pull/935 into smaller chunks.

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java [hunk 1]
```diff
@@ -160,19 +166,22 @@ public void enqueue(String accountId, String stream, Iterator<? extends StreamEv
    * @throws IOException if fails to get stream information
    * @throws IllegalArgumentException If the stream doesn't exists
    */
-  public void asyncEnqueue(String accountId, final String stream,
-                           Map<String, String> headers, ByteBuffer body, Executor executor) throws IOException {
+  public ListenableFuture<Long> asyncEnqueue(String accountId, final String stream,
```
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java]: How is it being used in async case? The server doesn't block for the result to return.

**Comment by chtyim** [comment]
Instead of returning, isn't it better to do it with a wrapped MetricsCollector?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java (hunk 1 above)
Comment (jguery) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java]: Basically the server would do the following:

```
ListenableFuture<Long> future = streamWriter.asyncEnqueue(accountId, stream, getHeaders(request, stream),
                                                              request.getContent().toByteBuffer(), asyncExecutor);
Futures.addCallback(future, new FutureCallback<Long>() {
      @Override
      public void onSuccess(Long result) {
        // Perform logic
      }

      @Override
      public void onFailure(Throwable t) {
        LOG.debug("Failed to write event asynchronously to Stream {}", stream);
      }
    }, asyncExecutor);
```

**Comment by chtyim** [comment]
Also, the current change misses the `appendFile` case. That's why I think wrapping the MetricsCollector is a better way.

**Comment by jguery** [comment]
I will close this PR and open a new one with the proposed changes.

**Pull request closed without merging**